x-pack/plugins/reporting/server/export_types/csv_searchsource/csv_searchsource.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IClusterClient } from '@kbn/core-elasticsearch-server';
+import { DataPluginStart } from '@kbn/data-plugin/server/plugin';
+import { DiscoverServerPluginStart } from '@kbn/discover-plugin/server';
+import { CsvGenerator } from '@kbn/generate-csv';
+import { CONTENT_TYPE_CSV } from '@kbn/generate-csv/src/constants';
+import { CancellationToken } from '@kbn/reporting-common';
+import { Writable } from 'stream';
+import {
+  CSV_JOB_TYPE,
+  LICENSE_TYPE_CLOUD_STANDARD,
+  LICENSE_TYPE_ENTERPRISE,
+  LICENSE_TYPE_GOLD,
+  LICENSE_TYPE_PLATINUM,
+  LICENSE_TYPE_TRIAL,
+} from '../../../common/constants';
+import { getFieldFormats } from '../../services';
+import { ExportType, ExportTypeSetupDeps, ExportTypeStartDeps } from '../common';
+import { decryptJobHeaders } from '../common/decrypt_job_headers';
+import { JobParamsCSV, TaskPayloadCSV } from './types';
+
+/*
+ * @TODO move to be within @kbn/reporitng-export-types
+ */
+
+export interface CsvSearchsourceExportTypeStartDeps extends ExportTypeStartDeps {
+  discover: DiscoverServerPluginStart;
+  data: DataPluginStart;
+  esClient: IClusterClient;
+}
+
+export class CsvSearchsourceExportType extends ExportType<JobParamsCSV, TaskPayloadCSV> {
+  id = 'csv_searchsource';
+  name = CSV_JOB_TYPE;
+  jobType = CONTENT_TYPE_CSV;
+  jobContentEncoding = 'base64' as const;
+  jobContentExtension = 'csv' as const;
+  validLicenses = [
+    LICENSE_TYPE_TRIAL,
+    LICENSE_TYPE_CLOUD_STANDARD,
+    LICENSE_TYPE_GOLD,
+    LICENSE_TYPE_PLATINUM,
+    LICENSE_TYPE_ENTERPRISE,
+  ];
+  declare startDeps: CsvSearchsourceExportTypeStartDeps;
+
+  constructor(...args: ConstructorParameters<typeof ExportType>) {
+    super(...args);
+    const logger = args[2];
+    this.logger = logger.get('csv-searchsource-export');
+  }
+
+  setup(setupDeps: ExportTypeSetupDeps) {
+    this.setupDeps = setupDeps;
+  }
+
+  start(startDeps: CsvSearchsourceExportTypeStartDeps) {
+    this.startDeps = startDeps;
+  }
+
+  public async getEsClient() {
+    const startDeps = await this.startDeps;
+    return startDeps.esClient;
+  }
+
+  public async getDataService() {
+    const startDeps = await this.startDeps;
+    return startDeps.data;
+  }
+
```

Comment:
The only public methods of the ExportType sub-classes should be `createJob` and `runTask`. That way we know all the things that export type instance objects are really used for, and we keep that down to a small number of things.